apps/dashboard/shared/components/design-system/filters/AddressFilter.tsx

```diff
@@ -0,0 +1,183 @@
+  try {
+    normalize(address);
+  } catch {
+    return false;
+  }
+
+  return address.endsWith(".eth") && address.slice(0, -4).length >= 3;
+};
+
+export function AddressFilter({
+  onApply,
+  currentFilter = "",
+  className,
+}: AddressFilterProps) {
+  const [tempAddress, setTempAddress] = useState<string>(currentFilter);
+  const [isResolving, setIsResolving] = useState<boolean>(false);
+  const [ensAddressError, setEnsAddressError] = useState<string | null>(null);
+
+  const isValidAddress =
+    tempAddress.trim() &&
+    (isAddress(tempAddress.trim()) || isEnsAddress(tempAddress.trim()));
+
+  const handleApply = async () => {
+    const trimmedAddress = tempAddress.trim();
+
+    if (!trimmedAddress) {
+      onApply(undefined);
+      return;
+    }
+
+    // If it's already a valid Ethereum address, use it directly
+    if (isAddress(trimmedAddress)) {
+      onApply(trimmedAddress);
+      return;
+    }
+
+    // If it's an ENS name, resolve it
+    if (isEnsAddress(trimmedAddress)) {
+      setIsResolving(true);
+      try {
+        const resolvedAddress = await getEnsAddress(publicClient, {
+          name: normalize(trimmedAddress),
+        });
+
+        if (resolvedAddress) {
+          onApply(resolvedAddress);
+        } else {
+          // ENS name doesn't resolve to an address
+          onApply(undefined);
+          setEnsAddressError("ENS name could not be resolved");
+        }
+      } catch (error) {
+        console.error("Error resolving ENS address:", error);
+        onApply(undefined);
+      } finally {
+        setIsResolving(false);
+      }
+    } else {
+      onApply(undefined);
+    }
+  };
+
+  const handleReset = () => {
+    setTempAddress("");
+    onApply(undefined);
+  };
+
+  const handleOpenChange = (open: boolean) => {
+    if (open) {
+      setTempAddress(currentFilter);
+    }
+  };
+
+  return (
+    <Popover onOpenChange={handleOpenChange}>
+      <PopoverTrigger asChild>
+        <button
+          aria-label="Filter by address"
+          className={cn(
+            "group flex cursor-pointer items-center rounded-sm border p-1 transition-colors",
+            "hover:border-highlight bg-surface-hover border-transparent",
+            currentFilter && "border-highlight bg-surface-hover",
+            className,
+          )}
+        >
+          <Filter className="text-primary size-3" />
+        </button>
+      </PopoverTrigger>
+      <PopoverContent
+        side="bottom"
+        align="start"
+        sideOffset={8}
+        avoidCollisions={true}
+        className={cn(
+          "border-border-contrast bg-surface-contrast z-50 w-[260px] rounded-lg border p-0 shadow-lg",
+        )}
+      >
+        <div className="flex flex-col gap-1 py-1">
+          {/* Header */}
+          <div className="flex items-center justify-between px-3 py-2">
+            <span className="text-secondary font-mono text-xs font-medium uppercase">
+              SEARCH ADDRESS
```

Comment:
hey yo bro, the goal to make the text like this and add a uppercase is because using accessibility voice it's better to retrieve information without the uppercase. Since the data is the same, we need to pass the same goals to all users ! 

Screen readers will read "Search Address" naturally rather than spelling out "S-E-A-R-C-H A-D-D-R-E-S-S". Visual styling remains unchanged. 

```suggestion
              Search Address
```

Reply:
You're right. One thing to notice though: "Some older screen readers (software that converts text to audio) and text-to-speech (TTS) software may read capitalized text letter-by-letter." - not every screen reader does that. 

Also: The Web Content Accessibility Guidelines (WCAG) does not have specific guidelines for all-caps text.

changing from `SEARCH ADDRESS` to `Search Address` is optimizing for the <0.1% of users. If accessibility is a real concern, then there are other more important thing to prioritize here.